src/net/i2p/crypto/eddsa/math/ed25519/Ed25519FieldElement.java

```diff
@@ -942,6 +942,77 @@ public FieldElement pow22523() {
         return multiply(t0);
     }
 
+    /**
+     * Constant-time conditional move. Well, actually it is a conditional copy.
+     * Logic is taken from the SUPERCOP implementation at:
+     *   https://github.com/floodyberry/supercop/blob/master/crypto_sign/ed25519/ref10/fe_cmov.c
+     *
+     * @param g the other field element.
+     * @param b must be 0 or 1, otherwise results are undefined.
+     * @return a copy of this if b == 0, or a copy of g if b == 1.
+     */
+    @Override
+    public FieldElement cmov(FieldElement g, int b) {
+        Ed25519FieldElement that = (Ed25519FieldElement) g;
+        // f0 ... f9 are a copy of this.t
+        int f0 = this.t[0];
+        int f1 = this.t[1];
+        int f2 = this.t[2];
+        int f3 = this.t[3];
+        int f4 = this.t[4];
+        int f5 = this.t[5];
+        int f6 = this.t[6];
+        int f7 = this.t[7];
+        int f8 = this.t[8];
+        int f9 = this.t[9];
+        // g0 ... g9 are a copy of that.t
+        int g0 = that.t[0];
+        int g1 = that.t[1];
+        int g2 = that.t[2];
+        int g3 = that.t[3];
+        int g4 = that.t[4];
+        int g5 = that.t[5];
+        int g6 = that.t[6];
+        int g7 = that.t[7];
+        int g8 = that.t[8];
+        int g9 = that.t[9];
+        // x0 ... x9 are XORs of this.t and that.t
+        int x0 = f0 ^ g0;
+        int x1 = f1 ^ g1;
+        int x2 = f2 ^ g2;
+        int x3 = f3 ^ g3;
+        int x4 = f4 ^ g4;
+        int x5 = f5 ^ g5;
+        int x6 = f6 ^ g6;
+        int x7 = f7 ^ g7;
+        int x8 = f8 ^ g8;
+        int x9 = f9 ^ g9;
```

Comment:
Non-blocking: each of these steps could possibly be rolled up, e.g.

```
for (int i = 0; i < 10; i++) {
    x[i] = t[i] ^ g[i];
}
```

See the equivalent handling in `add()`, `subtract()` and `negate()`. We'd need to check whether the branch predictor would interfere with the multiple sequential for loops, even though they would be fixed-length (unlike the for loop that this PR replaces, which conditionally ended on the secret `b`).

Even more concisely:

```
b = -b;
for (int i = 0; i < 10; i++) {
    x[i] = t[i] ^ g[i];
    x[i] &= b;
    f[i] ^= x[i];
}
```

But I'm less confident that the re-ordering of `b = -b` doesn't enable the compiler to make any optimisations.